common/util/rviz_plugins/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_display_base.hpp

```diff
@@ -0,0 +1,405 @@
+    m_display_twist_property{"Display Twist", true, "Enable/disable twist visualization", this},
+    m_display_predicted_paths_property{
+      "Display Predicted Paths", true, "Enable/disable predicted paths visualization", this},
+    m_display_path_confidence_property{
+      "Display Predicted Path Confidence", true, "Enable/disable predicted paths visualization",
+      this},
+    m_default_topic{default_topic}
+  {
+    // iterate over default values to create and initialize the properties.
+    for (const auto & map_property_it : detail::kDefaultObjectPropertyValues) {
+      const auto & class_property_values = map_property_it.second;
+      const auto & color = class_property_values.color;
+      // This is just a parent property to contain the necessary properties for the given class:
+      m_class_group_properties.emplace_back(
+        class_property_values.label.c_str(), QVariant(),
+        "Groups polygon properties for the given class", this);
+      auto & parent_property = m_class_group_properties.back();
+      // Associate a color and opacity property for the given class and attach them to the
+      // parent property of the class so they can have a drop down view from the label property:
+      m_polygon_properties.emplace(
+        std::piecewise_construct, std::forward_as_tuple(map_property_it.first),
+        std::forward_as_tuple(
+          QColor{color[0], color[1], color[2]}, class_property_values.alpha, &parent_property));
+    }
+    init_color_list(predicted_path_colors);
+  }
+
+  void onInitialize() override
+  {
+    RosTopicDisplay::RTDClass::onInitialize();
+    m_marker_common.initialize(this->context_, this->scene_node_);
+    this->topic_property_->setValue(m_default_topic.c_str());
+    this->topic_property_->setDescription("Topic to subscribe to.");
+  }
+
+  void load(const rviz_common::Config & config) override
+  {
+    RosTopicDisplay::Display::load(config);
+    m_marker_common.load(config);
+  }
+
+  void update(float wall_dt, float ros_dt) override { m_marker_common.update(wall_dt, ros_dt); }
+
+  void reset() override
+  {
+    RosTopicDisplay::reset();
+    m_marker_common.clearMarkers();
+  }
+
+  void clear_markers() { m_marker_common.clearMarkers(); }
+
+  void add_marker(visualization_msgs::msg::Marker::ConstSharedPtr marker_ptr)
+  {
+    m_marker_common.addMessage(marker_ptr);
+  }
+
+protected:
+  /// \brief Convert given shape msg into a Marker
+  /// \tparam ClassificationContainerT List type with ObjectClassificationMsg
+  /// \param shape_msg Shape msg to be converted
+  /// \param centroid Centroid position of the shape in Object.header.frame_id frame
+  /// \param orientation Orientation of the shape in Object.header.frame_id frame
+  /// \param labels List of ObjectClassificationMsg objects
+  /// \return Marker ptr. Id and header will have to be set by the caller
+  template <typename ClassificationContainerT>
+  boost::optional<Marker::SharedPtr> get_shape_marker_ptr(
```

Comment:
galactic supports C++17, so `std::optional` is better.

Reply:
It was replaced to std::optional on https://github.com/tier4/AutowareArchitectureProposal.iv/pull/574/commits/fd0b931837052c8c7d570e6e7bea66f050d9609b